src/Drivers/nlpMDSForm_raja_ex4.hpp

```diff
@@ -0,0 +1,218 @@
+#ifndef HIOP_EXAMPLE_RAJA_EX4
+#define HIOP_EXAMPLE_RAJA_EX4
+
+#include "hiopInterface.hpp"
+
+//this include is not needed in general
+//we use hiopMatrixDense in this particular example for convienience
+#include <hiopMatrixDense.hpp>
+#include <hiopLinAlgFactory.hpp>
+
+#ifdef HIOP_USE_MPI
+#include "mpi.h"
+#else
+#define MPI_COMM_WORLD 0
+#define MPI_Comm int
+#endif
+
+#include <cassert>
+#include <cstring> //for memcpy
+#include <cstdio>
+#include <cmath>
+
+/* Problem test for the linear algebra of Mixed Dense-Sparse NLPs
+ *  min   sum 0.5 {x_i*(x_{i}-1) : i=1,...,ns} + 0.5 y'*Qd*y + 0.5 s^T s
+ *  s.t.  x+s + Md y = 0, i=1,...,ns
+ *        [-2  ]    [ x_1 + e^T s]   [e^T]      [ 2 ]
+ *        [-inf] <= [ x_2        ] + [e^T] y <= [ 2 ]
+ *        [-2  ]    [ x_3        ]   [e^T]      [inf]
+ *        x <= 3
+ *        s>=0
+ *        -4 <=y_1 <=4, the rest of y are free
+ *        
+ * The vector 'y' is of dimension nd = ns (can be changed in the constructor)
+ * Dense matrices Qd and Md are such that
+ * Qd  = two on the diagonal, one on the first offdiagonals, zero elsewhere
+ * Md  = minus one everywhere
+ * e   = vector of all ones
+ *
+ * Coding of the problem in MDS HiOp input: order of variables need to be [x,s,y] 
+ * since [x,s] are the so-called sparse variables and y are the dense variables
+ */
+
+class Ex4 : public hiop::hiopInterfaceMDS
+{
+public:
+  Ex4(int ns_in, std::string mem_space)
+    : Ex4(ns_in, ns_in, mem_space)
+  {
+  }
+  
+  Ex4(int ns_in, int nd_in, std::string mem_space);
+
+  virtual ~Ex4();
+  
+  bool get_prob_sizes(long long& n, long long& m);
+
+  /**
+   * @todo will param _type_ live on host or device?
+   * @todo register pointers with umpire in case they need to be copied
+   * from device to host.
+   */
+  bool get_vars_info(const long long& n, double *xlow, double* xupp, NonlinearityType* type);
+
+  /**
+   * @todo fill out param descriptions below to determine whether or not
+   * they will reside on device and will have to be accessed/assigned to 
+   * in a RAJA kernel
+   *
+   * @param[out] m ?
+   * @param[out] clow ?
+   * @param[out] cupp ?
+   * @param[out] type ?
```

Comment:
I think these parameters live on host.

Reply:
I think they do, just as you described. We do need to document this correctly. Thanks for pointing this out.